### DISCUSSION TITLE
Does it support decorator metadata?

### DISCUSSION THREAD

**Discussion by ValentinGurkov** [rolldown/rolldown#451] [Q&A] [answered]
esbuild does not, so I wanted to ask.

**Comment by ValentinGurkov** [comment]
~~I think I answered my own question since it uses `esbuild` internally~~

**Comment by hyfdev** [comment, marked as answer]
Well
- Rolldown doesn't use `esbuild` internally.
- I just take a look at https://github.com/evanw/esbuild/issues/257 and we probably will follow the esbuild's behavior.

**Reply by ValentinGurkov** [reply]
My mistake, the site states that Vite uses `esbuild` internally not Rolldown itself. 
And thanks for clarifying.

**Comment by ValentinGurkov** [comment]
> Well
> 
> * Rolldown doesn't use `esbuild` internally.
> * I just take a look at [Support emitting typescript decorator metadata evanw/esbuild#257](https://github.com/evanw/esbuild/issues/257) and we probably will follow the esbuild's behavior.

**Reply by thomasfr** [reply]
According to https://rolldown.rs/guide/notable-features#built-in-transforms it does support decorator metadata:

> - TypeScript
>   - Sets configurations based on the tsconfig.json when the [tsconfig](https://rolldown.rs/reference/InputOptions.tsconfig) option is provided.
>   - Supported legacy decorators and decorator metadata.
